http3/server.go

```diff
@@ -237,21 +239,78 @@ func (s *Server) serveImpl(startListener func() (quic.EarlyListener, error)) err
 	}
 }
 
+func (s *Server) generateAltSvcHeader() {
+	s.altSvcGenerationErr = nil
+
+	// This code assumes that we will use protocol.SupportedVersions if no quic.Config is passed.
+	supportedVersions := protocol.SupportedVersions
+	if s.QuicConfig != nil && len(s.QuicConfig.Versions) > 0 {
+		supportedVersions = s.QuicConfig.Versions
+	}
+	var versionStrings []string
+	for _, version := range supportedVersions {
+		if v := versionToALPN(version); len(v) > 0 {
+			versionStrings = append(versionStrings, v)
+		}
+	}
+
+	extractPort := func(addr net.Addr) (int, error) {
+		_, portStr, err := net.SplitHostPort(addr.String())
+		if err != nil {
+			return 0, err
+		}
+
+		portInt, err := net.LookupPort("tcp", portStr)
```

Comment:
Well, there's technically a test which supplies ":https" as the port, and it is valid to run `http3.Server.ListenAndServe` with an address like ":https" given (which should launch a udp listener on 0.0.0.0:443). I don't really find it meaningful to support this and would rather have the function simply split by ':' and then do an `Atoi`, so that `SetQuicHeaders` doesn't return an error. But since it's already supported and `SetQuicHeaders` returns an error, doesn't make sense to change the API now :confused: